sdk/communication/AzureCommunicationCalling/Swift_Generated_Interfaces/AzureCommunicationCalling.h

```diff
@@ -0,0 +1,1124 @@
+    open var groupId: UUID!
+}
+
+open class TeamsMeetingCoordinatesLocator : AbstractJoinMeetingLocator {
+
+    public init!(with threadId: String!, organizerId: UUID!, tenantId: UUID!, messageId: String!)
+
+    
+    /// The thread identifier of meeting
+    open var threadId: String! { get }
+
+    
+    /// The organizer identifier of meeting
+    open var organizerId: UUID!
+
+    
+    /// The tenant identifier of meeting
+    open var tenantId: UUID!
+
+    
+    /// The message identifier of meeting
+    open var messageId: String! { get }
+}
+
+open class TeamsMeetingLinkLocator : AbstractJoinMeetingLocator {
+
+    public init!(meetingLink: String!)
+
+    
+    /// The link of the meeting
+    open var meetingLink: String! { get }
+}
+
+open class IncomingCallInformation : NSObject {
+
+    
+    /// Deallocates the memory occupied by this object.
+    open func dealloc()
+
+    
+    open var fromDisplayName: String! { get }
+
+    
+    open var hasIncomingVideo: Bool { get }
+
+    
+    // Class extension begins for IncomingCallInformation.
+    open var from: CommunicationIdentifier! { get }
+
+    open var to: CommunicationIdentifier! { get }
+
+    open var callId: UUID { get }
+
+    open class func from(payload: [AnyHashable : Any]!) -> IncomingCallInformation!
+}
+
+// Class extension ends for IncomingCallInformation.
+
+
+/// Options for creating CallAgent
+open class CallAgentOptions : NSObject {
+
+    public init!()
+
+    
+    /// Deallocates the memory occupied by this object.
+    open func dealloc()
+
+    
+    /// Specify the display name of the local participant for all new calls
+    open var displayName: String!
+}
+
+
+/// Call agent created by the CallClient factory method createCallAgent It bears the responsibility of managing calls on behalf of the authenticated user
+open class CallAgent : NSObject {
+
+    
+    /// Deallocates the memory occupied by this object.
+    open func dealloc()
+
+    
+    /// Returns the list of all active calls.
+    open var calls: [Call]! { get }
+
+    
+    /**
+     * The delegate that will handle events from the ACSCallAgent.
+     */
+    unowned(unsafe) open var delegate: CallAgentDelegate!
+
+    
+    open func handlePush(notification: IncomingCallInformation!, completionHandler: ((Error?) -> Void)!)
+
+    
+    open func unRegisterPushNotifications(completionHandler: ((Error?) -> Void)!)
+
+    
+    // Class extension begins for CallAgent.
+    open func startCall(participants: [CommunicationIdentifier]!, options: StartCallOptions!) -> Call!
+
+    open func join(with meetingLocator: AbstractJoinMeetingLocator!, joinCallOptions: JoinCallOptions!) -> Call!
+
+    open func registerPushNotifications(deviceToken: Data!, completionHandler: ((Error?) -> Void)!)
+}
+
+// Class extension ends for CallAgent.
+
+
+/// Describes a call
+open class Call : NSObject {
+
+    
+    /// Deallocates the memory occupied by this object.
```

Comment:
Okay, I take it back... **this** seems like the `CallClient`, not what you are currently calling the `CallClient` or the `CallAgent`. How does this sequence occur?

Reply:
CallClient is the main entry point which for each token/user can create multiple CallAgents.
CallAgent object can now be used to place outgoing calls, join group calls and when there is incoming calls are recieved. They are all Call objects which gets created and destroyed as and when there is a call going on.

Reply:
This has already been vetted through archboard for other languages, so it should be fine here too.